haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -349,6 +399,10 @@ def split(
             if split_by == "passage":
                 elements = text.split("\n\n")
             elif split_by == "sentence":
+                if self.add_page_number:
+                    # SentenceTokenizer will remove "\f" if it is at the end of a sentence, so substituting it in these
+                    # cases with [NEW_PAGE] to don't lose any page breaks.
+                    text = re.sub(r"([\.:?!])\f", r"\1 [NEW_PAGE]", text)
```

Comment:
we use this regex twice, better creating a single function (and unit tests)

Reply:
I moved this regex to a separate function and added a test for it.